testdata/src/a/a.go

```diff
@@ -13,4 +13,36 @@ func main() {
 	// With another comments
 	fmt.Printf("%d") // nolint // foobar
 	fmt.Printf("%d") // foobar // nolint
+
+	// With nolint directive above the block.
+	//nolint
+	fmt.Printf("%d")
+
+	// This statement has
+	// a lot of comments
+	// but in the end
+	// we mark it with the directive
+	//nolint
+	fmt.Printf("%d")
+
+	// This statement has
+	// a lot of comments
+	// and in the end
+	// we MISS mark it with
+	// the directive
+	fmt.Printf("%d") // want "Printf format %d reads arg #1, but call has 0 args"
+
+	// Multiline support
+	fmt.Printf( // want "Printf format %d reads arg #1, but call has 0 args"
+		"%d",
+	)
+
+	fmt.Printf( //nolint // multiline on diagnositcs line
+		"%d",
+	)
+
+	//nolint // multiline on top line
+	fmt.Printf(
+		"%d",
+	)
```

Comment:
💭  I want a test for a `//nolint` directive with a following empty line.

```suggestion
	)
	
	//nolint

	fmt.Printf( // want "Printf format %d reads arg #1, but call has 0 args"
		"%d",
	)
```

(Being too busy, I couldn't care about the specification which you meant.)

Reply:
This will actually fail because a comment, even with an empty line, belongs to the following node in the ast. I'm no expert on this area but I've noticed that the relationship between nodes and comments are weird sometimes. I also know that there's packages like [`dst`](https://github.com/dave/dst) which tries to improve this.

If you think this is important, I can have a look and ensure that there's no empty line between the comment and the node and manually separate them?

Reply:
No, I don't think that is a problem.
Just I didn't understand about the specification which you meant.

Reply:
💭
By the way, I think that a blank line often separates a block. (But ast is not so)
Thinking this is an issue, we can fix it. But now, it's too hard.
Now we should merge this PR.

Reply:
Oh, sorry, I don't know about any specification per se, I just know it's pretty common in both go (like `golangci-lint`) and other languages. It also expands the support to work for whole blocks like ranges and functions.

I don't think it's **that** hard to implement. We now have access to the last `*ast.CommentGroup` and `ast.Node`. We could basically just check that the `*ast.CommentGroup` Pos  must be on the same line or the line above as the `ast.Node` Pos. I can see if i'ts easy fixable!

Reply:
I actually agree that this should not be valid so I added support for this behaviour. PTAL.

Reply:
thanks, great job!